src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Apprentice/ApprenticesFilterModel.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Html;
+using SFA.DAS.CommitmentsV2.Types;
+using SFA.DAS.EmployerCommitmentsV2.Web.Extensions;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Models.Apprentice
+{
+    public class ApprenticesFilterModel
+    {
+        public int PageNumber { get; set; } = 1;
+        public string SearchTerm { get; set; }
+        public string SelectedProvider { get; set; }
+        public string SelectedCourse { get; set; }
+        public ApprenticeshipStatus? SelectedStatus { get; set; }
+        public DateTime? SelectedEndDate { get; set; }
+        public string SortField { get; set; }
+        public bool ReverseSort { get; set; }
+
+        public IEnumerable<string> ProviderFilters { get; set; } = new List<string>();
+        public IEnumerable<string> CourseFilters { get; set; } = new List<string>();
+        public IEnumerable<ApprenticeshipStatus> StatusFilters { get; set; } = new List<ApprenticeshipStatus>();
+        public IEnumerable<DateTime> EndDateFilters { get; set; } = new List<DateTime>();
+
+        private const int PageSize = Constants.ApprenticesSearch.NumberOfApprenticesPerSearchPage;
+        public int PagedRecordsFrom => TotalNumberOfApprenticeshipsFound == 0 ? 0 : (PageNumber - 1) * PageSize + 1;
+        public int PagedRecordsTo {
+            get
+            {
+                var potentialValue = PageNumber * PageSize;
+                return TotalNumberOfApprenticeshipsFound < potentialValue ? TotalNumberOfApprenticeshipsFound: potentialValue;
+            }
+        }
+        public bool ShowSearch => TotalNumberOfApprenticeships >= Constants.ApprenticesSearch.NumberOfApprenticesRequiredForSearch;
+
+        public bool SearchOrFiltersApplied => !string.IsNullOrWhiteSpace(SearchTerm)
+                                              || !string.IsNullOrWhiteSpace(SelectedProvider)
+                                              || !string.IsNullOrWhiteSpace(SelectedCourse)
+                                              || SelectedStatus.HasValue
+                                              || SelectedEndDate.HasValue;
+
+        public HtmlString FiltersUsedMessage => this.GetFiltersUsedMessage();
+
+        public int TotalNumberOfApprenticeships { get; set; }
+        public int TotalNumberOfApprenticeshipsFound { get; set; }
+        public int TotalNumberOfApprenticeshipsWithAlertsFound { get; set; }
+        
+        public IEnumerable<PageLink> PageLinks {
+            get
+            {
+                var links = new List<PageLink>();
```

Comment:
This seems like a lot of logic to be placed into a Get statement. Would this be better to either extract out into a helper and use that or break it down internally into private methods?

Reply:
I agree with you. Spent some time investigating and my conclusion is both of those options lead to other problems. helper needs injection somewhere - either into the model or the mapper, both look like a misuse of DI. putting into a method just moves the problem somewhere else. it's worth noting that the `PageLinks` property shares logic with `BuildSortRouteData()` function.